### PR TITLE
#7506 Proper handling of wildcards in FilterLayer

### PR DESCRIPTION
**Pull request by alexander-fedorenko**
## Description
Proper handling of wildcards in FilterLayer.
Added utility to process wildcards in the string based on operator used. "like" & "ilike" operators will get wildcards converted to the CQL format where "%" is used for multiple character substitution and "_" is used for a single character.
Following rules are applied:
- WPS requests will not have filter value unconditionally wrapped with ** anymore. There is a check if wildcard is presented in a value. Wrapping will happen only if there are no wildcards in the string.
- CQL: % and _ characters will be escaped with backslash before conversion.
- CQL: * and . will be converted into % and _ only in case if they are not prepended with !
- CQL: !* and !. will be converted into * and .
- CQL: value will be wrapped with %% only if no wildcards are passed into condition. "Te!*s!.t" will be wrapped with %% as there are no wildcards (both cases are escaped with !). "Te*s.t" won't be wrapped.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#7506

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/utils/FilterUtils.js [hunk 1]
```diff
@@ -834,21 +834,57 @@ export const cqlArrayField = function(attribute, operator, value) {
...
+ * All other operators will receive value as-is, no wildcard processing is applied
+ * @param value
+ * @param operator
+ * @returns {string}
+ */
+export const processCqlWildcards = function(value, operator) {
+    const escapedQuotes = escapeCQLStrings(value);
+    const startCharIsWildcard = value.startsWith('*');
+    const endCharIsWildcard = value.endsWith('*');
+    const startsWithWildcard = startCharIsWildcard && !endCharIsWildcard;
```
Comment (offtherailz) [inline web/client/utils/FilterUtils.js:850]: Starts or ends was only examples. Like filter can contain * everywhere.
We need to follow the same convention in the tooltip:

> use * for any number of any char
> use . for a single char
> use ! to escape above two (* and .) 

WFS requests in fact are sent using this convention (Try to use the advanced filter in feature grid):
```
<ogc:PropertyIsLike matchCase="false" wildCard="*" singleChar="." escapeChar="!">
```
The filter, when no wildcard is used is generated like that (inserting `| STATE_NAME | ilike | a |`)

```
<ogc:PropertyName>STATE_NAME</ogc:PropertyName><ogc:Literal>*a*</ogc:Literal>
```
image

Accordingly with the [reference](https://docs.geoserver.org/latest/en/user/styling/ysld/reference/filters.html) CQL uses `%` for multi-character and _ for single char. 

So we need to:

- [ ]  produce a CQL_FILTERS that transforms the string of the operators like and ilike like that: 
    - convert `*` and `.` with `%` and `_`
    - not convert when `!.` or `!*`
    - Escapes when `%` or `_" are present, ( Don't remember the escape char, I think `\.` and `\%` will work, but need to test)

Please document this accordingly in this method.

- [ ] I note right now that inserting `| STATE_NAME | ilike | a *|` in advanced filter of feature grid we have a filter like 
```
<ogc:PropertyName>STATE_NAME</ogc:PropertyName><ogc:Literal>*a**</ogc:Literal>
```

That inconsitent with what we declared in the issue. Therefore we should see to fix also this behavior.

**Review by offtherailz — CHANGES_REQUESTED** [review]
Please read my comment inline. Check when the wildcards are applied in the feature grid filter and propose a solution to make filters consistent.

File: web/client/utils/FilterUtils.js (hunk 1 above)
Comment (alexander-fedorenko) [reply web/client/utils/FilterUtils.js:859]: I think I misunderstood requirements, sorry.
All cases listed above should be covered with the changes applied by the last commit.

**Comment by alexander-fedorenko** [comment]
@offtherailz please review when it's convenient. I've updated PR description with the new rules applied to the filter value. Thanks!

**Review by offtherailz — APPROVED** [review]
:+1: LGTM

**Pull request merged**